### DISCUSSION TITLE
guests should be able to use "queue length" stat of ride to avoid long queues

### DISCUSSION THREAD

**Discussion by 733737** [OpenRCT2/OpenRCT2#16841] [Feature requests]
Peeps should be able to use the "queue length" statistic of the ride to determine whether they enter a queue. When they are about to enter a queue for a ride, there should be another check based on the queue length variable and the guest's happiness / tiredness. This way peeps wouldn't just enter a full queue because they want to ride something.

**Comment by ocalhoun6** [comment]
Maybe along the same lines, guests should be *more* likely to enter a queue if the line is very short.

I know, personally, if I'm going to a park (with free rides) and I see a ride with basically nobody in line, I'd be more likely to choose that ride, even if maybe it's not my favorite ride that I'm the most excited about going on, I might go on it anyway just because the 'cost' of getting on that ride is very low, so why not? So I'd be more likely to choose a ride outside my usual preferences if the line is extremely short and I can get on it right away. For example, I might think the Ferris Wheel is pretty boring, but if there's nobody in line and I can walk right on to it without waiting ... then maybe I'll go on it anyway just for the heck of it.

(Though this logic would be completely different in a pay-per-ride park. If it's pay-per-ride, then a short queue would be _far_ less important of a factor, because I still wouldn't want to spend money on a ride I don't really care for.)

----

While I do like this idea, it makes sense, and it maybe _should_ be introduced ... it also opens a whole can of worms in that the consequences of it will ripple outward through the game balance -- it's a much bigger and more significant change than it appears to be on the face of it.

Making guests less likely to enter a long queue means you'll have fewer guests in queues in general. That means:

-->More guests on paths
---->More overcrowding problems
------>More difficult to get a high guest count, especially in small maps that force you to have a compact park
---->More litter on paths
------>More handymen needed
-->Fewer guests waiting in queues
---->Rides that peeps don't want to ride in the rain more likely to run out of peeps in the queue before the rain ends
------>Covered rides that peeps will ride in the rain become more useful and valuable; non-covered rides become less useful and valuable
---->Queue line TVs and entertainers become slightly less useful
---->More guest happiness (assuming no overcrowding problems)
------>Easier to get a high park rating
---->_Consistency_ of ride throughput becomes more important. A long ride with one huge train will struggle to keep enough guests in the queue to fill the train, while a ride with many short trains will work better. Before this change, those two rides would perform roughly equally ... but after this change, the ride with many small trains will perform much better.
